FlowCrypt/Controllers/Threads/ThreadDetailsViewController.swift

```diff
@@ -220,6 +220,8 @@ extension ThreadDetailsViewController {
             handleMissedPassPhrase(for: rawMimeData, at: indexPath)
         case let .wrongPassPhrase(rawMimeData, passPhrase):
             handleWrongPassPhrase(for: rawMimeData, with: passPhrase, at: indexPath)
+        case let .keyMismatch(rawMimeData):
+            handleKeyMismatch(for: rawMimeData, at: indexPath)
```

Comment:
Please also add handlers for .keyMismatch and for .badMdc and .noMdc and also for general errors (not sure what those are called)

Reply:
Maybe we should do this in another PR?